staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/convert.go

```diff
@@ -248,7 +248,7 @@ func newExtensions(s *apiextensions.JSONSchemaProps) (*Extensions, error) {
 
 	if s.XPreserveUnknownFields != nil {
 		if !*s.XPreserveUnknownFields {
-			return nil, fmt.Errorf("'x-kubernetes-preserve-unknown-fields' must be true or undefined")
+			return nil, fmt.Errorf("internal error: 'x-kubernetes-preserve-unknown-fields' must be true or undefined")
```

Comment:
is this only run on persisted schemas that should already be valid?

Reply:
it's one of those checks which should never happen because we validate that already in the normal validation. It had the same error string, so I changed it here.